server/aws-lsp-notification/src/language-server/notificationServer.ts

```diff
@@ -0,0 +1,76 @@
+import {
+    InitializeParams,
+    InitializedParams,
+    HandlerResult,
+    PartialInitializeResult,
+    InitializeError,
+} from '@aws/language-server-runtimes/server-interface'
+import { ServerBase, ServerFeatures } from '@aws/lsp-core'
+import { NotificationService, ShowNotification } from './notificationService'
+import { NotificationFollowupParams, NotificationParams } from '@aws/language-server-runtimes/protocol'
+import { FilesystemMetadataStore } from '../notifications/metadata/filesystemMetadataStore'
+import { S3Fetcher } from '../notifications/toolkits/s3Fetcher'
+import { CritieriaFilteringFetcher } from '../notifications/toolkits/criteriaFilteringFetcher'
+import { MetadataFilteringFetcher } from '../notifications/metadata/metadataFilteringFetcher'
+import { Fetcher } from '../notifications/fetcher'
+
+export class NotificationServer extends ServerBase {
+    // Silently drop notifications from constructor to initliaze when ability to send notifications is available
```

Comment:
Why do you need to do this?

Reply:
Just TS minutiae.  It's not used before initialize is called, but it can't be set until initialized is called.  The other option is to make it optional and do non-null assertions everywhere.

Reply:
In general, the code here will likely change/evolve quite a bit as notifications is implemented.